src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -3335,3 +3339,27 @@ bool clang_c_convertert::is_field_global_storage(const clang::FieldDecl *field)
 
   return false;
 }
+
+bool clang_c_convertert::is_ConstructorOrDestructor(
+  const clang::FunctionDecl &fd)
+{
+  return fd.getKind() == clang::Decl::CXXConstructor ||
+         fd.getKind() == clang::Decl::CXXDestructor;
+}
+
+bool clang_c_convertert::check_member_expr_virtual_overriding(
+  const clang::Decl &decl)
+{
+  // It just can't happen in C
+  return false;
+}
+
+bool clang_c_convertert::get_vft_binding_expr(
+  const clang::MemberExpr &,
+  exprt &)
+{
+  log_error(
+    "MemberExpr call to virtual/overriding function cannot happen in C");
+  abort();
```

Comment:
do we need to abort here? why is this different from check_member_expr_virtual_overriding?

Reply:
Yep, it would be really weird if a C program contains: 
```
|   |     |-CXXMemberCallExpr 0x5649f692df60 <col:10, col:26> 'int'
|   |     | `-MemberExpr 0x5649f692df30 <col:10, col:13> '<bound member function type>' ->do_something 0x5649f6922cd8
```
 where `do_something 0x5649f6922cd8` points to a virtual function: 
```
| |-CXXMethodDecl 0x5649f6922cd8 <line:11:3, col:47> col:15 used do_something 'int ()' 
```
or an overriding function:
``
| |-CXXMethodDecl 0x5649f692b890 <line:17:5, col:49> col:17 used do_something 'int ()' virtual
| | |-Overrides: [ 0x5649f6922cd8 Bird::do_something 'int ()' ]
``